enhancements/ibm-public-cloud/ibm-public-cloud-support.md

```diff
@@ -0,0 +1,251 @@
+
+Enable OpenShift 4.x deployment on IBM Cloud.
+
+### Goals
+
+- Enable the deployment of OpenShift 4.x on IBM Public Cloud by providing the IBM
+  Public Cloud team the necessary tools to generate manifests needed for a hosted
+  control plane.
+- Ensure that this deployment model remains functional through regular e2e testing
+  on IBM Public Cloud.
+- Make the necessary product changes to make IBM Public Cloud a fully supported 
+  cloud provider.
+- Ensure that clusters running on IBM Public Cloud are conformant OpenShift clusters
+  (pass Kubernetes and OpenShift conformance e2e tests). Fully document any tests that
+  must be skipped due to differences from a traditional deployment.
+
+### Non-Goals
+
+- Make hosted control planes a supported deployment model outside of IBM Public Cloud.
+- Define the automation needed on the IBM Public Cloud side to provision new clusters.
+- Create an alternate installer for OpenShift.
+
+## Proposal
+
+Support for IBM Public Cloud will be added in 3 phases:
+
+1. Beta: Initial rollout. Enables users to create OpenShift clusters
+   using RHEL 7 worker nodes. Nodes are labeled as both worker and master to support
+   SLOs that have masters in their node selectors. IBM provides out of tree cloud support
+   for storage, service load balancers. Ingress, image registry, storage and auth are 
+   configured after initial provisioning (day 2). Calico is used for the SDN.
+2. GA: In-tree support for ingress, image registry, storage, auth, load balancers.
+3. Post-GA: Add support for RHCOS workers
+
+In more detail:
+
+### Beta
+
+Enables an OpenShift cluster to be hosted on top of a Kubernetes/OpenShift cluster. 
+
+Given a release image, a CLI tool generates manifests that instantiate the control plane
+components so that they can be applied to a namespace. IBM provides the automation
+that applies these manifests and instantiates worker nodes that add themselves to 
+the new cluster. Minting of kubelet certificates for these worker nodes is handled
+by IBM automation.
+
+Components that run on the management cluster include:
+- etcd\+
+- kubernetes apiserver
+- kubernetes controller manager
+- kubernetes scheduler
+- openshift apiserver
+- openshift controller manager
+- cluster version operator
+- control plane operator(s)\*
+- oauth server\+
+- vpn server\+
+
+\* - new component
+
+\+ - provisioned by IBM
+
+Once a control plane is running, the Kubernetes API server connects to a VPN server to
+which worker nodes also connect. This enables the API server to access the pod and service
+network of the worker nodes. This is needed for aggregated API services running on the
+worker nodes.
+
+The cluster version operator running on the management cluster connects to the Kubernetes
+API server via the service network and starts applying manifests for SLOs that must run
+on the worker nodes. These include (among others):
+
+- network (dns, multus)
+- ingress
+- image registry
+- OLM
+- samples
+- console
```

Comment:
The console node selector currently matches only master nodes. There is no technical reason the console can't run on workers, just that we wanted to avoid running it next to user workloads if possible since the console backend handles all logged-in user tokens. This will likely have to change, though.

cc @benjaminapetersen